editor/components/inserter/helpers/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Helpers associated with the Inserter Component.
+ *
+ * This file contains a group of functions that help in managing the inserter.
+ */
+
+export const isShownBlock = state => block => block.title.toLowerCase().indexOf( state.filterValue.toLowerCase() ) !== -1;
+
+export const listVisibleBlocksByCategory = ( state ) => ( blocks ) => {
+	return blocks.reduce( ( groups, block ) => {
+		if ( ! isShownBlock( state )( block ) ) {
+			return groups;
+		}
+		if ( ! groups[ block.category ] ) {
+			groups[ block.category ] = [];
+		}
+		groups[ block.category ].push( block );
+		return groups;
+	}, {} );
+};
+
+const flattenObjectIntoArray = ( object ) => {
+	let list = [];
+	for ( const key in object ) {
+		list.push( object[ key ] );
+	}
+	list = list.reduce( ( values, objectValue ) => values.concat( objectValue ), [] );
+
+	return list;
+};
+
+const nextFocusableRef = ( component ) => {
```

Comment:
Instead of having a signature like this `component` => `block` which is not clear to me because the component should implement a specific interface for the function to work properly. I'd prefer something like `( currentBlock, blocks)` => `block`.

Also, should we test these functions?